### PR TITLE
Make docker run locally

### PR DESCRIPTION
**Pull request by TracerDS**
This PR localizes Docker. It no longer copies mtasa repo but local copy of the mta.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -21,6 +21,15 @@ RUN apt-get update && \
...
 
+RUN mkdir -p Client Server Shared utils vendor
+
+COPY Client Client
+COPY Server Server
+COPY Shared Shared
+COPY utils utils
+COPY vendor vendor
+COPY ./* .
+
```
Comment (TheNormalnij) [inline Dockerfile:32]: You should use `-v` parameter in `docker run`  to mount these folders into a container

File: utils/docker-entrypoint.sh [hunk 2]
```diff
@@ -21,11 +21,6 @@ if [[ $AS_BUILDAGENT = "1" ]]; then
 fi
 
 # Manually invoke build process
-# So, first make a shallow clone of the repository if it not exists
-umask 000
-if [ ! -f ./premake5.lua ]; then
-    git clone --depth=1 https://github.com/multitheftauto/mtasa-blue.git .
-fi
```
Comment (TheNormalnij) [inline utils/docker-entrypoint.sh:28]: Why? Nothing is wrong here. This line allows you to build MTA without cloning the MTA repository.

Comment (TracerDS) [reply utils/docker-entrypoint.sh:28]: this line actually clones the entire repo all the time.

File: docker-build.sh [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+NAME="mtasa"
+
+docker build . -t $NAME
+ID = $(docker images -q $NAME)
```
Comment (TheNormalnij) [inline docker-build.sh:6]: Why do you need that and how to use it?

Comment (TracerDS) [reply docker-build.sh:6]: When you execute the script, it will build the docker image and print its id

File: utils/docker-entrypoint.sh (hunk 2 above)
Comment (TheNormalnij) [reply utils/docker-entrypoint.sh:28]: It doesn't. Check your command line and current folder

**Comment by TheNormalnij** [comment]
It works locally now

```sh
docker build -t mtasabuilder:latest .
docker run --rm -v ./:/build mtasabuilder:latest
```

File: Dockerfile (hunk 1 above)
Comment (TracerDS) [reply Dockerfile:32]: How would it work?

File: utils/docker-entrypoint.sh (hunk 2 above)
Comment (TracerDS) [reply utils/docker-entrypoint.sh:28]: It does. Docker pulls mta repo from github if that line is saved. Without it, it runs only local changes

Comment (TheNormalnij) [reply utils/docker-entrypoint.sh:28]: No. I tested it

File: Dockerfile (hunk 1 above)
Comment (TheNormalnij) [reply Dockerfile:32]: You should mount mta repository (or an empty folder) to `/build` folder in a container. `docker-entrypoint.sh` runs inside this folder.

File: docker-build.sh (hunk 3 above)
Comment (TheNormalnij) [reply docker-build.sh:6]: We don't need that ID usually

File: utils/docker-entrypoint.sh (hunk 2 above)
Comment (TracerDS) [reply utils/docker-entrypoint.sh:28]: Me too. It didnt accept local copy unless I remove that line.
If you try to build mta then you already have necessary files in your filesystem.

File: docker-build.sh (hunk 3 above)
Comment (TracerDS) [reply docker-build.sh:6]: Its needed when you want to fully terminate a docker session

Comment (TheNormalnij) [reply docker-build.sh:6]: Container tags are used for that.
```sh
docker stop <tag>
docker rm <tag>
```

File: utils/docker-entrypoint.sh (hunk 2 above)
Comment (TheNormalnij) [reply utils/docker-entrypoint.sh:28]:
```sh
cd <mta sa repo here>
docker build -t mtasabuilder:latest .
docker run --rm -v ./:/build mtasabuilder:latest
```
Does it work?

File: docker-build.sh (hunk 3 above)
Comment (TracerDS) [reply docker-build.sh:6]: both of these commands require the id that is provided via the output of the command above

File: utils/docker-entrypoint.sh (hunk 2 above)
Comment (TracerDS) [reply utils/docker-entrypoint.sh:28]: "volume name is too short, names should be at least two alphanumeric characters."

**Comment by botder** [comment]
Did you ever try the steps under **Building with Docker** from the README?
https://github.com/multitheftauto/mtasa-blue/blob/550a152551ec101d042ae774f3a54df2b04bea85/README.md?plain=1#L113-L122

**Comment by TracerDS** [comment]
> Did you ever try the steps under **Building with Docker** from the README?
> https://github.com/multitheftauto/mtasa-blue/blob/550a152551ec101d042ae774f3a54df2b04bea85/README.md?plain=1#L113-L122

I did, but no matter what I do I always get a copy of the github repo instead of my local one.

**Comment by botder** [comment]
> I did, but no matter what I do I always get a copy of the github repo instead of my local one.

Alright, but did you try using `--entrypoint /bin/bash` to look around and see if you properly mounted the files in /build ?

**Comment by TheNormalnij** [comment]
> "volume name is too short, names should be at least two alphanumeric characters."

Do you use powershell in windows?

**Comment by TracerDS** [comment]
> Alright, but did you try using `--entrypoint /bin/bash` to look around and see if you properly mounted the files in /build ?

Didn't try that exact method but when `ls`ing the root dir, no directories were copied.

> Do you use powershell in windows?

Nope

**Comment by botder** [comment]
> > Alright, but did you try using `--entrypoint /bin/bash` to look around and see if you properly mounted the files in /build ?
> 
> Didn't try that exact method but when `ls`ing the root dir, no directories were copied.

You're supposed to mount your local directory into the Docker container, and by poking around within a /bin/bash shell, you can see if your mount is actually working. If it's not there, then obviously the docker-entrypoint.sh script is going to clone the repository everytime.

File: utils/docker-entrypoint.sh (hunk 2 above)
Comment (TheNormalnij) [reply utils/docker-entrypoint.sh:28]:
```
docker run --rm -v <full path to repo>:/build mtasabuilder:latest
```
?

**Comment by TheNormalnij** [comment]
Invalid / a system issue

**Pull request closed without merging**